packages/react/src/components/Button/Button.types.ts

```diff
@@ -341,6 +341,12 @@ export interface IButtonProps
    * overall SplitButton container
    */
   primaryActionButtonProps?: IButtonProps;
+
+  /**
+   * Optional props to avoid nested roles in split button.
+   * @see https://github.com/microsoft/fluentui/issues/18484
+   */
+  disableNestedRole?: boolean;
```

Comment:
I just chatted w/ the team -- there's actually an accessibility concern with this approach: until we rewrite splitbutton to actually just be two buttons, we need to keep all the current roles on there, even though they go against the spec.

The parent role is needed b/c that's the element that gets keyboard focus, and the child roles are both needed for touch-based screen readers, or those wouldn't be able to expose both the primary action and the dropdown.

I think the team is going to close this PR, and follow up with closing their original bug, since the practical impact of the nested menuitems in this case should be minimal.